include/boost/geometry/srs/projections/spar.hpp

```diff
@@ -41,195 +41,104 @@ namespace spar
 
 // Static parameters holder
 
-#define BOOST_GEOMETRY_PROJECTIONS_DETAIL_TYPENAME_PX \
-typename P0, typename P1, typename P2, typename P3, typename P4, \
-typename P5, typename P6, typename P7, typename P8, typename P9, \
-typename P10, typename P11, typename P12, typename P13, typename P14
+template <typename ...>
+class parameters {};
 
-#define BOOST_GEOMETRY_PROJECTIONS_DETAIL_PX \
-P0, P1, P2, P3, P4, P5, P6, P7, P8, P9, P10, P11, P12, P13, P14
-
-#ifndef DOXYGEN_NO_DETAIL
 namespace detail
 {
 
-template <BOOST_GEOMETRY_PROJECTIONS_DETAIL_TYPENAME_PX>
-struct map_params_to_cons
+// TODO: implement this as a sequence utility
+template <std::size_t I, typename ...>
+struct parameters_base
 {
-  typedef boost::tuples::cons
-    <
-        P0,
-        typename map_params_to_cons
-            <
-                P1, P2, P3, P4, P5, P6, P7, P8, P9, P10, P11, P12, P13, P14,
-                boost::tuples::null_type
-            >::type
-    > type;
+    BOOST_GEOMETRY_STATIC_ASSERT_FALSE(
+        "I out of bounds.",
```

Comment:
Can this be a `std` static assert? But it's TODO anyway so you can leave it for now

Reply:
It is std `static_assert` which fails when this template is instantiated. This is a utility I added recently which allows to do this. Note that raw `static_assert(false)` would fail always when the class is compiled, even if the template is not instantiated.

The TODO comment is only about making it a metafunction and moving to util/sequence.hpp in the future.